src/stdlib_bitsets.fypp

```diff
@@ -1170,8 +1170,8 @@ module stdlib_bitsets
 !! Version: experimental
 !!
 !! Used to define assignment for `bitset_large`.
-            type(bitset_large), intent(out) :: set1
-            type(bitset_large), intent(in)  :: set2
+            type(bitset_large), intent(inout) :: set1
```

Comment:
Why is this needed?
And if it is needed, I guess the corresponding interfaces should be also modified for `bitset_64`.
It might be wise to open another PR with these changes only related to `stdlib_bitset`

Reply:
Should the corresponding specs be modified too?

Reply:
This change and the change you mentioned below have been done to fix unexpected behavior when assigning the `bitset_large` type.

Without this fix, the tests for sorting `bitset_large` in the following environment will fail:
- msys2-build (MSYS, x86_64) in ci_windows.yml
- msys2-build (MINGW32, i686) in ci_windows.yml
- msys2-build (MINGW32, i686) in ci_windows.yml
- Build (macos-latest, 11, cmake) in CI.yml
- Build (macos-latest, 12, cmake) in CI.yml

As you said, opening a new issue for further discussion is better.

---
I share the reason for those modification.

The component `blocks` of `set1` with the `intent(out)` attribute is not allocated at the beginning of the assignment procedure. Therefore, when `size( set2 % blocks, kind=bits_kind )` is executed, an error occurs that `blocks` have not been allocated. To avoid this error, I modified the `intent` attribute from `intent(out)` to `intent(inout)`.

Although I could not find a direct description of this situation in the Fortran Standard, the following statements are relevant:

>The INTENT attribute for an allocatable dummy argument applies to both the allocation status and the
definition status. An actual argument that corresponds to an INTENT (OUT) allocatable dummy argument
is deallocated on procedure invocation (9.7.3.2). 

> When a variable of derived type is deallocated, any allocated allocatable subobject is deallocated. If an error condition occurs during deallocation, it is processor dependent whether an allocated allocatable subobject is deallocated.

> The INTENT (OUT) attribute for a nonpointer dummy argument specifies that the dummy argument becomes undefined on invocation of the procedure, except for any subcomponents that are default-initialized (7.5.4.6).

An error still occurred in the assignment procedure after the modification.
`allocate( set1 % blocks( size( set2 % blocks, kind=bits_kind ) ) )` gives the error "`set1%blocks` is already allocated" when the same variable of the `bitset_large` type was specified on both sides of the assignment operator. This sometimes happens when the array indices for the bitset array to be sorted are the same.
Since `set1` and `set2` are the same when the same array element is now specified on both sides of the assignment operator, it is natural that this error occurs.
Therefore, the automatic allocation functionality is used instead of explicit allocation.

Similar modifications are not required for `type(bitset_64)`. This is because `type(bitset_64)` uses an integer corresponding `integer(int64)` for storing bitset and thus uses a built-in assignment operation.

Reply:
Ok. Thank you for the explanation. A new PR with a test replicating this issue would be useful.
Otherwise, this PR looks good to me.

Reply:
I followed your comment and opened another issue #726 with replication instructions because discussing it in more detail would be better.